src/annotator/config/index.js

```diff
@@ -1,12 +1,57 @@
 import settingsFrom from './settings';
 import { toBoolean } from '../../shared/type-coercions';
 
+/**
+ * Total list of configuration keys.
+ */
+const configurationKeys = [
+  'annotations',
+  'assetRoot',
+  'branding',
+  'clientUrl',
+  'enableExperimentalNewNoteButton',
+  'experimental',
+  'group',
+  'focus',
+  'theme',
+  'usernameUrl',
+  'onLayoutChange',
+  'openSidebar',
+  'query',
+  'requestConfigFromFrame',
+  'services',
+  'showHighlights',
+  'notebookAppUrl',
+  'sidebarAppUrl',
+  'subFrameIdentifier',
+  'externalContainerSelector',
+];
+
+/**
+ * List of allowed configuration keys per application context. Keys omitted
+ * in a given context will be removed from the relative configs when calling
+ * getConfig.
+ */
+const configurationContexts = {
+  // For testing
+  all: [...configurationKeys],
+
+  annotator: [...configurationKeys],
```

Comment:
It might not be that difficult to list just the relevant keys for each context here rather than using the complete list for all contexts and then just removing one of the items in the `notebook` context.

Reply:
Thats the long term plan, but if it's okay with you, id like to land this unchanged from how it works now. I still want to   fix the things outlined in the ticket and then circle back to this and come up with an audited list of config settings that we agree the annotator needs.  This is also a risker change than just the structure of how the config gets created.